src/main/resources/dbreplica/migration/V1.0.10__configure_replication_plsql.sql

```diff
@@ -0,0 +1,183 @@
+--
+-- Copies a task record into the history table.
+--
+CREATE OR REPLACE FUNCTION cft_task_db.add_task_history(l_task cft_task_db.tasks, l_is_delete boolean)
+ RETURNS bigint
+ LANGUAGE plpgsql
+AS $function$
+declare
+  l_update_id bigint;
+  l_new_task boolean;
+  l_result_action varchar default 'ignore';
+  l_update_lock integer;
+begin
+
+  insert into cft_task_db.task_history
+  (task_id, task_name, task_type, due_date_time,
+   state, task_system, security_classification,
+   title, major_priority, minor_priority, assignee,
+   auto_assigned, execution_type_code, work_type,
+   role_category, has_warnings, assignment_expiry,
+   case_id, case_type_id, case_category, case_name,
+   jurisdiction, region, location, business_context,
+   termination_reason, created, updated_by, updated,
+   update_action)
+  values
+  (l_task.task_id, l_task.task_name, l_task.task_type, l_task.due_date_time,
+   l_task.state, l_task.task_system, l_task.security_classification,
+   l_task.title, l_task.major_priority, l_task.minor_priority, l_task.assignee,
+   l_task.auto_assigned, l_task.execution_type_code, l_task.work_type,
+   l_task.role_category, l_task.has_warnings, l_task.assignment_expiry,
+   l_task.case_id, l_task.case_type_id, l_task.case_category, l_task.case_name,
+   l_task.jurisdiction, l_task.region, l_task.location, l_task.business_context,
+   l_task.termination_reason, l_task.created, l_task.last_updated_user, l_task.last_updated_timestamp,
+   l_task.last_updated_action)
+  returning update_id into l_update_id;
+
```

Comment:
Maybe there ought to be a commit here if it's all correctly inserted into the history?

Reply:
As discussed with Ravi, this will be refactored RWA-2085